Weaken assertion to allow additional elements in calling convention

clementbera opened the pull request:
Fixed assertion to make ShadowStack entries work in SOMns.

[smarr edit]: this weakens the assertion in the specializer to allow for additional elements in the calling convention, which do not need to be represented by nodes.

clementbera committed:
fixed assertion

smarr commented:
@clementbera could you apply a fix also for checkstyle? It moved to github.
Something like the following should work: https://github.com/smarr/SOMns/commit/92021a111e6fd22c08210e273547ca2ece24372b

clementbera committed:
updated checkstyle dependency to the fancier new version

smarr committed:
Update .checkstyle_checks.xml from SOMns

Checkstyle changed its config format somewhat.
This is basically a copy from SOMns. Style is not change with intention.

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr reviewed (approved):
Looks good to me.

Thank you!
